### PR TITLE
Added segment colors to indicate traffic severity

### PR DESCRIPTION
**Pull request by ashleycoxley**
- Traffic for each segment scaled to reflect a percentage of 8 cyclists (a reasonable guess at the 90th percentile; can be automatically generated in future)
- Color scaling computed using HSL, then converted to RGB and finally hex
- Segment color lines change when traffic is updated
- Default color before the first traffic update is green, indicating no traffic

@harperj

### CODE REVIEW

File: static/js/trailTraffic.js [hunk 1]
```diff
@@ -27,6 +28,23 @@ var Segment = function(data, map) {
...
+    var high_value = 8;
+    var valuePCT = (parseInt(thisSegment.trafficCount(), 10) / high_value);
+    if (valuePCT > 1) {
+      valuePCT = 1;
+    }
+    // Conversion function takes a scaled hue [0,1]
+    var hue = ((1-valuePCT) * 120) / 360;
+    // Saturation and lightness set to produce red-green scale
+    var rgb = hslToRgb(hue, 1, 0.7);
+    var hex = '#' + RGBToHex(rgb[0], rgb[1], rgb[2]);
```
Comment (harperj) [inline static/js/trailTraffic.js:44]: This is fine, but you can use use RGB and HSL colors in CSS so you could at least skip the hex conversion step...

File: redwood_segment_info.json [hunk 2]
```diff
@@ -1 +1 @@
...vation_gain": 53.8, "country": "United States", "created_at": "2014-05-20T06:27:12Z", "maximum_grade": 21.1, "resource_state": 3, "starred": false, "start_latitude": 37.807893, "activity_type": "Ride"}, {"athlete_count": 574, "updated_at": "2016-12-04T09:03:00Z", "private": false, "elevation_low": 431.8, "end_latitude": 37.815846148878336, "id": 2000123, "athlete_segment_stats": {"pr_date": null, "effort_count": 0, "pr_elapsed_time": null}, "city": "Oakland", "elevation_high": 453.4, "star_count": 0, "start_longitude": -122.17543476261199, "state": "CA", "average_grade": 5.9, "map": {"resource_state": 3, "polyline": "ilxeFnkuhVGBCLQLKBQNUAg@FU@OGYAc@BQB_@LWAo@HQJQ@WJEHc@PELq@z@", "id": "s2000123"}, "end_longitude": -122.17647730372846, "hazardous": false, "start_latlng": [37.81333677470684, -122.17543476261199], "end_latlng": [37.815846148878336, -122.17647730372846], "effort_count": 1362, "distance": 364.79, "climb_category": 0, "name": "Roberts to Redwood Bowl", "total_elevation_gain": 21.6, "country": "United States", "created_at": "2012-08-06T04:57:45Z", "maximum_grade": 11.7, "resource_state": 3, "starred": false, "start_latitude": 37.81333677470684, "activity_type": "Ride"}]}
```
Comment (harperj) [inline redwood_segment_info.json:1]: 😡

**Review by harperj — APPROVED** [review]

File: static/js/trailTraffic.js (hunk 1 above)
Comment (ashleycoxley) [reply static/js/trailTraffic.js:44]: the color is passed to the Google Maps API, which only takes hex

Comment (harperj) [reply static/js/trailTraffic.js:44]: 👍

**Comment by harperj** [comment]
🚢  it

**Pull request merged**